azure-eventhubs-samples/src/main/java/com/microsoft/azure/eventhubs/samples/Benchmarks/IngressBenchmark.java

```diff
@@ -0,0 +1,149 @@
+ */
+package com.microsoft.azure.eventhubs.samples.Benchmarks;
+
+import com.microsoft.azure.eventhubs.EventData;
+import com.microsoft.azure.eventhubs.EventHubClient;
+import com.microsoft.azure.servicebus.ConnectionStringBuilder;
+import com.microsoft.azure.servicebus.ServiceBusException;
+
+import java.io.IOException;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.function.BiConsumer;
+import java.util.function.Consumer;
+
+/*
+ * Performance BenchMark is specific to customers load pattern!!
+ *
+ * This sample is intended to highlight various variables available to tune latencies
+ *
+ * Two variables that cannot be exercised in the below Code - are
+ * 1) Network proximity
+ *      (make sure to run in same region/AzureDataCenter as your target scenario - to get identical results)
+ * 2) Throughput units configured on the EventHubs namespace
+ *      Latency numbers can only be measured when the target load is <= allowed load
+ *      if the Target load (no. of messages per sec or no. of bytes transferred per sec) exceeds the Throughput unit configuration
+ *      then you might see a long tail in 999'ile of ~4 secs - which service intentionally delays to slow down the sender
+ *
+ * If you are running this against an EventHubs Namespace in a shared public instance
+ * - results might slightly vary across runs; if you want more predictable results - use Dedicated EventHubs.
+ *
+ * This program generates 10000 samples of "latency per send call in milliseconds"
+ * After the completion of this program - copy the output to an excel sheet & crunch the numbers (like avg, %'s etc)
+ */
+public class IngressBenchmark {
+
+    public static void main(String[] args)
+            throws ServiceBusException, ExecutionException, InterruptedException, IOException {
+
+        final String namespaceName = "----ServiceBusNamespaceName-----";
+        final String eventHubName = "----EventHubName-----";
+        final String sasKeyName = "-----SharedAccessSignatureKeyName-----";
+        final String sasKey = "---SharedAccessSignatureKey----";
+        final ConnectionStringBuilder connStr = new ConnectionStringBuilder(namespaceName, eventHubName, sasKeyName, sasKey);
+
+
+        // ***************************************************************************************************************
+        // List of variables involved
+        // 1 - EVENT SIZE
+        // 2 - NO OF CONCURRENT SENDS per sec
+        // 3 - NO OF EVENTS - CLIENTS CAN BATCH & SEND <-- and there by optimize on ACKs returned from the Service (typically, this number is supposed to help bring 2 down)
+        // 4 - NO OF SENDERS PER CONNECTION <-- This sample uses 1 Sender AMQP link per 1 Connection
```

Comment:
No such variable? Should probably be discussed separately from the list of variables if you're not going to provide a knob.